src/bijectors/named_bijector.jl

```diff
@@ -0,0 +1,31 @@
+struct NamedBijector{Bs} <: AbstractBijector
```

Comment:
I just thought of the following three things.
1. What is the advantage of a `NamedBijector` over a named tuple of `Bijector`s?
2. In the function signatures you check that the names of the tuple of bijector and the tuple of inputs are identical - however, that is not a strict requirement for the implementation to work it seems, the names of the inputs could be just a (possibly permuted) subset of the names of the bijectors. Do you actually want to disallow this possibility?
3. If you actually want to dispatch on the names of the bijectors, maybe it would be helpful to define
   ```julia
    struct NamedBijector{names,B<:NamedTuple{names}}
        bijectors::B
    end
    ```
    Then one could dispatch on `names` only, e.g.,
    ```julia
    @generated function (b::NamedBijector{names})(x::NamedTuple{names}) where {names}
        return :(($([:($n = b.bs.$n(x.$n)) for n in names]...), ))
    end
    ```

Reply:
1. Fair point, though I think it's useful in case we might want to define special behaviors when using named tuples. Say we need something like the `NamedCouplingLayer` bijector described above, we can define something like a `AbstractNamedBijector{names}` and have `NamedBijector` and `NamedCouplingLayer` be concrete subtypes. Then we can define compositions and such between `AbstractNamedBijector`s rather than on a case-by-case basis.
2.  Good point! :+1: I'll change that for sure.
3. Good idea:) And I think it would be a good idea to put this in a `AbstractNamedBijector{names}` too, as mentioned in (1).

Reply:
I think I've now addressed these comments :+1: Thanks!